docs/spec.md

```diff
@@ -160,10 +163,56 @@ following constraints:
 * (C12) If `quantization_dimension` is not empty, then
   `0 <= quantization_dimension`.
 
-At the moment, `QuantizationScale` is a floating-point constant, but there is
-strong interest in integer-based scales, represented with multipliers and
-shifts. We are planning to explore this in the near future
-([#1404](https://github.com/openxla/stablehlo/issues/1404)).
+In order to allow operation using only integer arithmetic, the floating-point
+`scale`, `S`, is realized using integer `multipler`, `M` and `shift` value,
+`n >= 0`, such that `round_nearest_integer(S) = (M + round)* 2^-n`, where
```

Comment:
I think it may be better to express the formula as:

M = round_nearest_integer(S*(2^n))

For example, if S=0.25 then we have M=(1<<30) and n=32. Then round_nearest_integer(S)=0 giving both sides as zero. round_nearest_inter(S*(2^n)) = (1<<30) =M gives a non zero comparison.

Reply:
That is a good point @dominicsymes. 

I agree with your point and incorporated in the spec. PTAL.
I note that, in order to maintain consistency with the rounding mode used in other quantized ops, I have used round_nearest_even consistently. The specification here demonstrates one way of using the rounding mode, other should be applicable as well.

Reply:
Thanks @sdasgup3 for the update. I agree that round_nearest_even() is more consistent. The example code could also be updated to use this (I've added a comment below).